packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -388,13 +388,8 @@ abstract contract MetaTxModule is BaseModule
         if (gasSettings.token == address(0)) {
             transactCall(wallet, feeRecipient, gasCost, "");
         } else {
-            bytes memory txData = abi.encodeWithSelector(
-                ERC20(0).transfer.selector,
-                feeRecipient,
-                gasCost
-            );
             require(
-                abi.decode(transactCall(wallet, gasSettings.token, 0, txData), (bool)),
+                transactCallTokenTransfer(wallet, gasSettings.token, feeRecipient, gasCost),
```

Comment:
I forgot this is why we previously didn't check the return value here.

At this point the relayer will have spent the gas anyway, so even if the transfer actually fails for old tokens the cost has already happened for him. So I don't think it's super important to check this here.

But we can still do it quite easily without a special `transactCallTokenTransfer` like this (copied from https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/SafeERC20.sol to be safe):
```
if (returndata.length > 0) { // Return data is optional
    require(abi.decode(returndata, (bool)), "SafeERC20: ERC20 operation did not succeed");
}
```

if `success == false` `BaseWallet.transact` will revert so we don't need to have access to it here.